### ISSUE TITLE
[Bug]: Honkai Star Rail and Genshin Crash

### ISSUE THREAD

**Issue by Sokyu04** [PlayCover/PlayCover#1401]
### Describe the bug

both genshin impact and honkai star rail crash when opened

### Steps to reproduce

using Playcover nightly, open the app and it crashes, Sonoma 14.3.1

### Expected behaviour

open properly

### Crash log

```shell
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               hkrpg [2506]
Path:                  /Users/USER/*/Honkai: Star Rail.app/hkrpg
Identifier:            com.HoYoverse.hkrpgoversea
Version:               2.0.0 (2)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-02-22 22:54:20.0140 -0500
OS Version:            macOS 14.3.1 (23D60)
Report Version:        12
Anonymous UUID:        A7810E7A-32D9-DD76-0E6F-D35B63BEAEBF


Time Awake Since Boot: 510 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000001, 0x000000019a11e0f8

Termination Reason:    Namespace SIGNAL, Code 5 Trace/BPT trap: 5
Terminating Process:   exc handler [2506]

Application Specific Signatures:
The code signature is not valid: The operation couldn’t be completed. (OSStatus error -67030.)

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_secinit.dylib       	       0x19a11e0f8 _libsecinit_appsandbox.cold.9 + 60
1   libsystem_secinit.dylib       	       0x19a11d1b8 _libsecinit_appsandbox + 1608
2   libsystem_trace.dylib         	       0x18d5e76d8 _os_activity_initiate_impl + 64
3   libsystem_secinit.dylib       	       0x19a11cb1c _libsecinit_initializer + 80
4   libSystem.B.dylib             	       0x19a133694 libSystem_initializer + 260
5   dyld                          	       0x18d521a24 invocation function for block in dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const::$_0::operator()() const + 168
6   dyld                          	       0x18d567328 invocation function for block in dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const + 340
7   dyld                          	       0x18d55a668 invocation function for block in dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const + 496
8   dyld                          	       0x18d5012fc dyld3::MachOFile::forEachLoadCommand(Diagnostics&, void (load_command const*, bool&) block_pointer) const + 300
9   dyld                          	       0x18d5596a0 dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const + 192
10  dyld                          	       0x18d566e3c dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const + 516
11  dyld                          	       0x18d51db38 dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const + 524
12  dyld                          	       0x18d527928 dyld4::PrebuiltLoader::runInitializers(dyld4::RuntimeState&) const + 44
13  dyld                          	       0x18d543360 dyld4::APIs::runAllInitializersForMain() + 84
14  dyld                          	       0x18d505fa0 dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*) + 3192
15  dyld                          	       0x18d504edc start + 1844


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000060   x1: 0x000000014a7043cc   x2: 0x0000000000000000   x3: 0x000000016b52b340
    x4: 0xffffffffdf1d906c   x5: 0x0000000000000020   x6: 0x000000016b52b2c0   x7: 0x0000000000000bb0
    x8: 0x00000001e318d9a8   x9: 0xdae43d95733e00ac  x10: 0x0000000000000200  x11: 0x0000000000000060
   x12: 0x0000000000000000  x13: 0x00000000000007fd  x14: 0x000000008845f810  x15: 0x0000000000000010
   x16: 0x000000018d69bcc0  x17: 0x00000001ed1dfbc8  x18: 0x0000000000000000  x19: 0x000000014a70438c
   x20: 0x000000016b52b320  x21: 0x00000000000001f5  x22: 0x0000000000000000  x23: 0x00000000000003e8
   x24: 0x000000019a11ed5e  x25: 0x000000019a11edae  x26: 0x0000000000000000  x27: 0x0000000000000001
   x28: 0x000000019a11edf2   fp: 0x000000016b52b2e0   lr: 0x816600019a11e0ec
    sp: 0x000000016b52b2c0   pc: 0x000000019a11e0f8 cpsr: 0x60001000
   far: 0x0000000000000000  esr: 0xf2000001 (Breakpoint) brk 1

Binary Images:
       0x1052f8000 -        0x1054d3fff io.playcover.PlayTools (1.1.7) <bfe88cab-fb54-3853-97a8-f01b6fee888a> /Users/USER/Library/Frameworks/PlayTools.framework/PlayTools
       0x1048d0000 -        0x1048d7fff com.HoYoverse.hkrpgoversea (2.0.0) <0ef87105-662e-3785-8be5-ec6f7ceefdef> /Users/USER/Library/Containers/io.playcover.PlayCover/*/com.HoYoverse.hkrpgoversea.app/hkrpg
       0x19a11c000 -        0x19a11effb libsystem_secinit.dylib (*) <0a37b0c4-518f-343b-9542-299958362705> /usr/lib/system/libsystem_secinit.dylib
       0x18d5e0000 -        0x18d5fafff libsystem_trace.dylib (*) <d243654e-bd35-337c-9714-b0b462fa20ca> /usr/lib/system/libsystem_trace.dylib
       0x19a132000 -        0x19a133fff libSystem.B.dylib (*) <68092300-d766-369d-8407-bf255e29e70f> /usr/lib/libSystem.B.dylib
       0x18d4ff000 -        0x18d593387 dyld (*) <50746901-db0e-39a0-b391-baaa6b82ad0f> /usr/lib/dyld
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???
       0x18d69a000 -        0x18d6d0fff libsystem_malloc.dylib (*) <faa92b9b-9273-3c30-bb9e-36346eca2faf> /usr/lib/system/libsystem_malloc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=1.6G resident=0K(0%) swapped_out_or_unallocated=1.6G(100%)
Writable regions: Total=542.5M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=542.5M(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                   256K        1 
Kernel Alloc Once                   32K        1 
MALLOC                           530.2M       13 
MALLOC guard page                   96K        6 
STACK GUARD                       56.0M        1 
Stack                             8176K        1 
__AUTH                            3182K      492 
__AUTH_CONST                      34.8M      718 
__CTF                               824        1 
__DATA                            16.0M      708 
__DATA_CONST                      38.2M      728 
__DATA_DIRTY                      2600K      293 
__FONT_DATA                          4K        1 
__LINKEDIT                       895.7M        3 
__OBJC_RO                         71.2M        1 
__OBJC_RW                         2171K        1 
__TEXT                           745.1M      748 
dyld private memory               4384K        4 
shared memory                       32K        2 
===========                     =======  ======= 
TOTAL                              2.4G     3723 



-----------
Full Report
-----------

{"app_name":"hkrpg","timestamp":"2024-02-22 22:54:20.00 -0500","app_version":"2.0.0","slice_uuid":"0ef87105-662e-3785-8be5-ec6f7ceefdef","build_version":"2","platform":6,"bundleID":"com.HoYoverse.hkrpgoversea","share_with_app_devs":1,"is_first_party":0,"bug_type":"309","os_version":"macOS 14.3.1 (23D60)","roots_installed":0,"name":"hkrpg","incident_id":"05C1CD33-8074-4032-A6DD-159D1D035501"}
{
  "uptime" : 510,
  "procRole" : "Default",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "Mac14,6",
  "coalitionID" : 2030,
  "osVersion" : {
    "train" : "macOS 14.3.1",
    "build" : "23D60",
    "releaseType" : "User"
  },
  "captureTime" : "2024-02-22 22:54:20.0140 -0500",
  "codeSigningMonitor" : 1,
  "incident" : "05C1CD33-8074-4032-A6DD-159D1D035501",
  "pid" : 2506,
  "translated" : false,
  "cpuType" : "ARM-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2024-02-22 22:54:19.9666 -0500",
  "procStartAbsTime" : 12265220332,
  "procExitAbsTime" : 12266346902,
  "procName" : "hkrpg",
  "procPath" : "\/Users\/USER\/*\/Honkai: Star Rail.app\/hkrpg",
  "bundleInfo" : {"CFBundleShortVersionString":"2.0.0","CFBundleVersion":"2","CFBundleIdentifier":"com.HoYoverse.hkrpgoversea","DTAppStoreToolsBuild":"15C5500a"},
  "storeInfo" : {"deviceIdentifierForVendor":"3BAF113B-0C42-5F2B-97DD-3820FDE05C57","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "com.HoYoverse.hkrpgoversea",
  "crashReporterKey" : "A7810E7A-32D9-DD76-0E6F-D35B63BEAEBF",
  "codeSigningID" : "com.HoYoverse.hkrpgoversea",
  "codeSigningTeamID" : "",
  "codeSigningFlags" : 570425857,
  "codeSigningValidationCategory" : 10,
  "codeSigningTrustLevel" : 4294967295,
  "instructionByteStream" : {"beforePC":"8wMCqvQDAKrhAwD5AgAAkEJIPJEBAIFSEgEAlGiDJPAIoSaRFE0AqQ==","atPC":"IAAg1H8jA9X\/wwDR\/XsCqf2DAJEIdSOQCHFF+QgBQPmogx\/4CABA+Q=="},
  "sip" : "enabled",
  "exception" : {"codes":"0x0000000000000001, 0x000000019a11e0f8","rawCodes":[1,6879830264],"type":"EXC_BREAKPOINT","signal":"SIGTRAP"},
  "termination" : {"flags":0,"code":5,"namespace":"SIGNAL","indicator":"Trace\/BPT trap: 5","byProc":"exc handler","byPid":2506},
  "os_fault" : {"process":"hkrpg"},
  "asiSignatures" : ["The code signature is not valid: The operation couldn’t be completed. (OSStatus error -67030.)"],
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":26652,"threadState":{"x":[{"value":96},{"value":5543838668},{"value":0},{"value":6095549248},{"value":18446744073157841004},{"value":32},{"value":6095549120},{"value":2992},{"value":8105023912,"symbolLocation":8,"symbol":"gCRAnnotations"},{"value":15772799507051184300},{"value":512},{"value":96},{"value":0},{"value":2045},{"value":2286286864},{"value":16},{"value":6667484352,"symbolLocation":0,"symbol":"free"},{"value":8273132488},{"value":0},{"value":5543838604},{"value":6095549216},{"value":501},{"value":0},{"value":1000},{"value":6879833438},{"value":6879833518},{"value":0},{"value":1},{"value":6879833586}],"flavor":"ARM_THREAD_STATE64","lr":{"value":9324140085397020908},"cpsr":{"value":1610616832},"fp":{"value":6095549152},"sp":{"value":6095549120},"esr":{"value":4060086273,"description":"(Breakpoint) brk 1"},"pc":{"value":6879830264,"matchesCrashFrame":1},"far":{"value":0}},"queue":"com.apple.main-thread","frames":[{"imageOffset":8440,"symbol":"_libsecinit_appsandbox.cold.9","symbolLocation":60,"imageIndex":2},{"imageOffset":4536,"symbol":"_libsecinit_appsandbox","symbolLocation":1608,"imageIndex":2},{"imageOffset":30424,"symbol":"_os_activity_initiate_impl","symbolLocation":64,"imageIndex":3},{"imageOffset":2844,"symbol":"_libsecinit_initializer","symbolLocation":80,"imageIndex":2},{"imageOffset":5780,"symbol":"libSystem_initializer","symbolLocation":260,"imageIndex":4},{"imageOffset":141860,"symbol":"invocation function for block in dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const::$_0::operator()() const","symbolLocation":168,"imageIndex":5},{"imageOffset":426792,"symbol":"invocation function for block in dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const","symbolLocation":340,"imageIndex":5},{"imageOffset":374376,"symbol":"invocation function for block in dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const","symbolLocation":496,"imageIndex":5},{"imageOffset":8956,"symbol":"dyld3::MachOFile::forEachLoadCommand(Diagnostics&, void (load_command const*, bool&) block_pointer) const","symbolLocation":300,"imageIndex":5},{"imageOffset":370336,"symbol":"dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const","symbolLocation":192,"imageIndex":5},{"imageOffset":425532,"symbol":"dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const","symbolLocation":516,"imageIndex":5},{"imageOffset":125752,"symbol":"dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const","symbolLocation":524,"imageIndex":5},{"imageOffset":166184,"symbol":"dyld4::PrebuiltLoader::runInitializers(dyld4::RuntimeState&) const","symbolLocation":44,"imageIndex":5},{"imageOffset":279392,"symbol":"dyld4::APIs::runAllInitializersForMain()","symbolLocation":84,"imageIndex":5},{"imageOffset":28576,"symbol":"dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*)","symbolLocation":3192,"imageIndex":5},{"imageOffset":24284,"symbol":"start","symbolLocation":1844,"imageIndex":5}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4381966336,
    "CFBundleShortVersionString" : "1.1.7",
    "CFBundleIdentifier" : "io.playcover.PlayTools",
    "size" : 1949696,
    "uuid" : "bfe88cab-fb54-3853-97a8-f01b6fee888a",
    "path" : "\/Users\/USER\/Library\/Frameworks\/PlayTools.framework\/PlayTools",
    "name" : "PlayTools",
    "CFBundleVersion" : "1"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4371316736,
    "CFBundleShortVersionString" : "2.0.0",
    "CFBundleIdentifier" : "com.HoYoverse.hkrpgoversea",
    "size" : 32768,
    "uuid" : "0ef87105-662e-3785-8be5-ec6f7ceefdef",
    "path" : "\/Users\/USER\/Library\/Containers\/io.playcover.PlayCover\/*\/com.HoYoverse.hkrpgoversea.app\/hkrpg",
    "name" : "hkrpg",
    "CFBundleVersion" : "2"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6879821824,
    "size" : 12284,
    "uuid" : "0a37b0c4-518f-343b-9542-299958362705",
    "path" : "\/usr\/lib\/system\/libsystem_secinit.dylib",
    "name" : "libsystem_secinit.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6666715136,
    "size" : 110592,
    "uuid" : "d243654e-bd35-337c-9714-b0b462fa20ca",
    "path" : "\/usr\/lib\/system\/libsystem_trace.dylib",
    "name" : "libsystem_trace.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6879911936,
    "size" : 8192,
    "uuid" : "68092300-d766-369d-8407-bf255e29e70f",
    "path" : "\/usr\/lib\/libSystem.B.dylib",
    "name" : "libSystem.B.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6665793536,
    "size" : 607112,
    "uuid" : "50746901-db0e-39a0-b391-baaa6b82ad0f",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  },
  {
    "size" : 0,
    "source" : "A",
    "base" : 0,
    "uuid" : "00000000-0000-0000-0000-000000000000"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6667476992,
    "size" : 225280,
    "uuid" : "faa92b9b-9273-3c30-bb9e-36346eca2faf",
    "path" : "\/usr\/lib\/system\/libsystem_malloc.dylib",
    "name" : "libsystem_malloc.dylib"
  }
],
  "sharedCache" : {
  "base" : 6665043968,
  "size" : 4065345536,
  "uuid" : "c32b200e-cf99-3583-bc9f-b7fb11382e8f"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=1.6G resident=0K(0%) swapped_out_or_unallocated=1.6G(100%)\nWritable regions: Total=542.5M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=542.5M(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nActivity Tracing                   256K        1 \nKernel Alloc Once                   32K        1 \nMALLOC                           530.2M       13 \nMALLOC guard page                   96K        6 \nSTACK GUARD                       56.0M        1 \nStack                             8176K        1 \n__AUTH                            3182K      492 \n__AUTH_CONST                      34.8M      718 \n__CTF                               824        1 \n__DATA                            16.0M      708 \n__DATA_CONST                      38.2M      728 \n__DATA_DIRTY                      2600K      293 \n__FONT_DATA                          4K        1 \n__LINKEDIT                       895.7M        3 \n__OBJC_RO                         71.2M        1 \n__OBJC_RW                         2171K        1 \n__TEXT                           745.1M      748 \ndyld private memory               4384K        4 \nshared memory                       32K        2 \n===========                     =======  ======= \nTOTAL                              2.4G     3723 \n",
  "legacyInfo" : {
  "threadTriggered" : {
    "queue" : "com.apple.main-thread"
  }
},
  "logWritingSignature" : "ed281f395a353ee53a0e719c9f0346ffd37cbf2b",
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "6410af69ed1e1e7ab93ed169",
      "factorPackIds" : {

      },
      "deploymentId" : 240000011
    },
    {
      "rolloutId" : "6112d17137f5d11121dcd4e2",
      "factorPackIds" : {

      },
      "deploymentId" : 240000504
    }
  ],
  "experiments" : [

  ]
}
}

Model: Mac14,6, BootROM 10151.81.1, proc 12:8:4 processors, 32 GB, SMC 
Graphics: Apple M2 Max, Apple M2 Max, Built-In
Display: Color LCD, 3456 x 2234 Retina, Main, MirrorOff, Online
Memory Module: LPDDR5, Hynix
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4388), wl0: Dec 15 2023 23:28:34 version 23.10.765.20.41.51.129 FWID 01-9b7ed8b7
AirPort: 
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: USB31Bus
USB Device: USB31Bus
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
```

### What version of PlayCover are you using?

Nightly/beta (661)

### What version of macOS are you using?

Sonoma (macOS 14.3.1)

### Issue Language

- [X] Yes my issue is written in English

**Comment by Sokyu04** [comment]
Playchain data must be cleared everytime I want to open the apps

**Comment by TheMoonThatRises** [comment]
Set the application type of the app to something (such as "public.app-category.games") to resign the app (this can be done under the app's settings -> misc).

**Comment by kamui545** [comment]
> Set the application type of the app to something (such as "public.app-category.games") to resign the app (this can be done under the app's settings -> misc).

Did not work for me, it still crashes and I have to clear Playchain data as well every time

**Comment by Sunhuahai** [comment]
> Set the application type of the app to something (such as "public.app-category.games") to resign the app (this can be done under the app's settings -> misc).
It workde for me, thank you!

**Comment by Sunhuahai** [comment]
> Set the application type of the app to something (such as "public.app-category.games") to resign the app (this can be done under the app's settings -> misc).

It worked for me, thank you!

**Comment by mannkeithc** [comment]
I installed PlayCover and Genshin Impact iOS on an M2 MacBook Pro, and was experiencing the same problem, where the game would initially crash producing a crash report but would run if I pressed the "Reopen" button on the crash report. I tried reassigning the application type under app's settings -> Misc to "public.app-category.games", but that made no difference. I finally resolved it by unchecking "Enable PlayChain (Experimental)" under app's settings -> Bypasses, and it now runs without generating a crash report.
